### PR TITLE
refactor(evm): Deprecate engine plugin + allow custom `consensus.Engine`

### PR DESCRIPTION
**Pull request by itsdevbear**
Allows custom `consensus.Engine`

### Summary by CodeRabbit

- Refactor: The software's architecture has been significantly restructured to improve maintainability and flexibility. This includes changes to the naming and organization of key components, such as the transition from "engine" to "comet" and the introduction of the "Polaris" struct.
- New Feature: The system now supports optional parameters in the `New` function, providing users with more customization options.
- Improvement: The system's logging functionality has been enhanced, with a new `logger` field added to the `Polaris` struct.
- Test: Additional test coverage has been added for the `cosmos/x/evm/plugins/historical` package, improving the reliability of the software.
- Chore: The Go language version used in the project has been updated from `1.21.1` to `1.21.3`, ensuring the software benefits from the latest language features and improvements.
  
Please note that these changes are primarily under-the-hood improvements and may not result in noticeable changes for end-users. However, they lay the groundwork for more robust and flexible software in the future.

### CODE REVIEW

File: cosmos/runtime/runtime.go [hunk 1]
```diff
@@ -21,123 +21,150 @@
...
-		panic(err)
-	}
+// StartServices starts the services of the Polaris struct.
+func (p *Polaris) StartServices() error {
+	go func() {
+		// TODO: these values are sensitive due to a race condition in the json-rpc ports opening.
+		// If the JSON-RPC opens before the first block is committed, hive tests will start failing.
+		// This needs to be fixed before mainnet as its ghetto af. If the block time is too long
+		// and this sleep is too short, it will cause hive tests to error out.
+		time.Sleep(5 * time.Second) //nolint:gomnd // as explained above.
```
Comment (calbera) [inline cosmos/runtime/runtime.go:155]: for now could just make this sleep time a function of the block time (like `2 * p.config.BlockTime` or so)

Comment (itsdevbear) [reply cosmos/runtime/runtime.go:155]: gonna fix properly next PR

File: eth/eth.go [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+// It takes a client type, configuration, host chain, consensus engine, and log handler
+// as parameters. It returns a pointer to the ExecutionLayer and an error if any.
+func New(
+	client string, cfg any, host core.PolarisHostChain,
+	engine consensus.Engine, logHandler log.Handler,
+) (*ExecutionLayer, error) {
+	clientFactories := map[string]func(
+		any, core.PolarisHostChain, consensus.Engine, log.Handler,
+	) (*ExecutionLayer, error){
+		"geth": newGethExecutionLayer,
```
Comment (ocnc) [inline eth/eth.go:101]: Could make the different clients an enum to feel less magic variable

**Review by ocnc — APPROVED** [review]
utack

**Pull request merged**